<launch>

  <node pkg="ess_imu_driver" type="ess_imu_driver_node" name="ess_imu_driver_node"
	output="screen">

    <!-- Frame ID for IMU topic. -->
    <param name="frame_id" type="string" value="imu_link" />

    <!-- The name of the serial port to read IMU data from. -->
    <param name="serial_port" type="string" value="/dev/ttyUSB0" />

    <!-- ext_sel: These bits select the function of GPIO2 terminal to be GPIO2,
   External Counter Reset Input, or External Trigger Input.
   The time_correction parameter below requires ext_sel = 1 (External Counter Reset Input)
   and a 1PPS signal connected to GPIO2
	 0: GPIO2
	 1: External Counter Reset Input
	 2: External Trigger Input -->
    <param name="ext_sel" type="int" value="1" />
    <!-- ext_pol: Selects the polarity of the External Counter Reset Input or
	 External Trigger Input function.
	 0: Positive Logic (Rising Edge)
	 1: Negative Logic (Falling Edge) -->
    <param name="ext_pol" type="int" value="0" />

    <!-- dout_rate: Specifies the data output rate.
	 The following lists the data output rate option with the recommended number of filter taps.

	 Samples per sec.    Recommended Filter
	 0x00 :2000Sps       TAP>=0
	 0x01 :1000Sps       TAP>=2
	 0x02 :500Sps        TAP>=4
	 0x03 :250Sps        TAP>=8
	 0x04 :125Sps        TAP>=16
	 0x05 :62.5Sps       TAP>=32
	 0x06 :31.25Sps      TAP>=64
	 0x07 :15.625Sps     TAP=128
	 0x08 :400 Sps       TAP>=8
	 0x09 :200 Sps       TAP>=16
	 0x0A :100 Sps       TAP>=32
	 0x0B :80 Sps        TAP>=32
	 0x0C :50 Sps        TAP>=64
	 0x0D :40 Sps        TAP>=64
	 0x0E :25 Sps        TAP=128
	 0x0F :20 Sps        TAP=128 -->
    <param name="dout_rate" type="int" value="9" />

    <!-- Specifies the type of filter (moving average filter and FIR Kaiser filter) and TAP setting.
	 For the FIR Kaiser filter, these bits also selects the cutoff frequency fc in Hz.

	 0: Moving average filter TAP=0
	 1: Moving average filter TAP=2
	 2: Moving average filter TAP=4
	 3: Moving average filter TAP=8
	 4: Moving average filter TAP=16
	 5: Moving average filter TAP=32
	 6: Moving average filter TAP=64
	 7: Moving average filter TAP=128
	 8: FIR Kaiser filter (parameter=8) TAP=32 and fc=50
	 9: FIR Kaiser filter (parameter=8) TAP=32 and fc=100
	 10: FIR Kaiser filter (parameter=8) TAP=32 and fc=200
	 11: FIR Kaiser filter (parameter=8) TAP=32 and fc=400
	 12: FIR Kaiser filter (parameter=8) TAP=64 and fc=50
	 13: FIR Kaiser filter (parameter=8) TAP=64 and fc=100
	 14: FIR Kaiser filter (parameter=8) TAP=64 and fc=200
	 15: FIR Kaiser filter (parameter=8) TAP=64 and fc=400
	 16: FIR Kaiser filter (parameter=8) TAP=128 and fc=50
	 17: FIR Kaiser filter (parameter=8) TAP=128 and fc=100
	 18: FIR Kaiser filter (parameter=8) TAP=128 and fc=200
	 19: FIR Kaiser filter (parameter=8) TAP=128 and fc=400 -->
    <param name="filter_sel" type="int" value="9" />

    <!-- Enables or disables the following sensor outputs. For models that support quaternion output, qtn_out can be set 1 = enable -->
    <param name="flag_out" type="int" value="1" />
    <param name="temp_out" type="int" value="1" />
    <param name="gyro_out" type="int" value="1" />
    <param name="accel_out" type="int" value="1" />
    <param name="qtn_out" type="int" value="0" />
    <param name="count_out" type="int" value="1" />
    <param name="checksum_out" type="int" value="1" />

    <!-- For attitude/quaternion output sets attitude motion profile
    # 0: modeA (standard)
    # 1: modeB (vehicle/high)
    # 2: modeC (construction/low) -->
    <param name="atti_profile" type="int" value="0" />

    <!-- Corrects the timestamp using the reset count field in the IMU message.
   This function is experimental.
   This function requires the above parameter ext_sel = 1 (External Counter Reset)
   This assumes that the IMU is pulsed with GNSS 1PPS on GPIO2, when the system time is beginning a new second. -->
    <param name="time_correction" type="int" value="0" />

    <!-- Remap topic to /imu/data_raw. -->
    <remap from="epson_imu/data_raw" to="imu/data_raw"/>
		<remap from="epson_imu/data" to="imu/data"/>
  </node>

</launch>
